mrc/localization/locator.py

```diff
@@ -1,36 +1,41 @@
 import threading
-from dynaconf import settings
-import itertools
 import time
+from typing import Callable
 
 
 class Locator(threading.Thread):
-    def __init__(self):
+    def __init__(self, read_data: Callable, process_data: Callable, interval: float):
         threading.Thread.__init__(self)
-        self.locations = dict(itertools.product(settings.get('ROBOTS'), [None]))
```

Comment:
I'd keep it but pass list of robots as constructor argument, so that we know about which robots we should ask process_data.